### PR TITLE
Make gherkin and messages peer dependencies

### PR DESCRIPTION
**Pull request by aurelien-reeves**
### 🤔 What's changed?

`@cucumber/gherkin` and `@cucumber/messages` are now peer dependencies. `@cucumber/messages-streams` has been removed from the dependencies as it seemed to be useless.

That should help into validating acceptance criteria of `@cucumber/gherkin` with the help of `@cucumber/gherkin-streams` without circular dependencies issues.
It may also prevent the need for releasing `@cucumber/gherkin-streams` as often as gherkin itself.

### ⚡️ What's your motivation? 

While trying to remove gherkin-streams from the monorepo, I faced some difficulties to update validation of acceptance criteria for gherkin as it depends on gherkin-streams which depends on gherkin.

Having a peer dependency rather than a hard one should help using gherkin-streams with working copy of gherkin.

### 🏷️ What kind of change is this?

- :boom: Breaking change (will cause existing user behaviour to not
  work it did before)

### CODE REVIEW

**Comment by aurelien-reeves** [comment]
@davidjgoss I am not sure for messages and message-streams
Should we let those as regular dependencies?

**Comment by davidjgoss** [comment]
I think the versions number as you have them with the `^` is good.

Also I think the plan is to remove the dependency on messages from gherkin so I think that concern will go away sooner or later?

**Comment by aurelien-reeves** [comment]
> I think the versions number as you have them with the `^` is good.

Great, thanks

> Also I think the plan is to remove the dependency on messages from gherkin so I think that concern will go away sooner or later?

Yeap, true. And at the moment, this is not really a concern 😅

Thanks for your answer!

**Review by davidjgoss — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,8 +12,15 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 ### Changed
 
 * `@cucumber/gherkin`, `@cucumber/messages` and `@cucumber/message-streams` are now
-  peer dependencies. You will have to explicitely depend on those in your projects
-  in addition to `@cucumber/gherkin-streams`.
+  peer dependencies. You now have to add `@cucumber/gherkin` in your dependencies:
+  ```diff
```
Comment (davidjgoss) [inline CHANGELOG.md:16]: Nice, didn't know about this pattern for diffs!

**Pull request merged**